File: longest_substring/longest_substring.py
def longest_substring(s):
    seen = set()
    L = 0
    maxLen = 0
    for R in range (len(s)):
        while s[R] in seen:
            seen.remove(s[L])
            L += 1
        seen.add(s[R])
        maxLen = max(maxLen, R - L + 1)
    return maxLen

s = "abcabcbb"
print("The longest substring length is :", longest_substring(s))

File: longest_substring/longest_substring.ts
export {};

function longestSubString(s: string): number {
  let seen = new Set();
  let L = 0;
  let maxLength = 0;
  for (let R = 0; R < s.length; R++) {
    while (seen.has(s[R])) {
      seen.delete(s[L]);
      L++;
    }
    seen.add(s[R]);
    maxLength = Math.max(maxLength, R - L + 1);
  }
  return maxLength;
}

const s = "abcabcbb";
console.log("The longest substring length is :", longestSubString(s));
